src/esp/sensor/Sensor.cpp

```diff
@@ -67,18 +88,28 @@ void Sensor::setTransformationFromSpec() {
   node().rotateZ(Magnum::Rad(spec_->orientation[2]));
 }
 
-void SensorSuite::add(const Sensor::ptr& sensor) {
-  const std::string uuid = sensor->specification()->uuid;
-  sensors_[uuid] = sensor;
+SensorSuite::SensorSuite(scene::SceneNode& node)
+    : Magnum::SceneGraph::AbstractFeature3D{node} {}
+
+void SensorSuite::add(sensor::Sensor& sensor) {
+  const std::string uuid = sensor.specification()->uuid;
+  sensors_.emplace(uuid, std::ref(sensor));
 }
 
 void SensorSuite::merge(SensorSuite& sensorSuite) {
   sensors_.insert(sensorSuite.getSensors().begin(),
                   sensorSuite.getSensors().end());
 }
 
-Sensor::ptr SensorSuite::get(const std::string& uuid) const {
-  return (sensors_.at(uuid));
+void SensorSuite::remove(sensor::Sensor& sensor) {
```

Comment:
`const sensor::Sensor&`.

Check the other places, see if there are more cases that need `const`.